explorer/templates.go

```diff
@@ -151,6 +151,58 @@ func float64Formatting(v float64, numPlaces int, useCommas bool, boldNumPlaces .
 	return []string{integer, dec[:places], dec[places:], trailingZeros}
 }
 
+func amountAsDecimalPartsTrimmed(v, numPlaces int64, useCommas bool) []string {
+
+	// Filter numPlaces to only allow up to 8 decimal places trimming (eg. 1.12345678)
+	if numPlaces > 8 {
+		numPlaces = 8
+	}
+
+	// Separate values passed in into int.dec parts.
+	intpart := v / 1e8
+	decpart := v % 1e8
+
+	// Format left side.
+	left := strconv.FormatInt(intpart, 10)
+	rightWithTail := fmt.Sprintf("%08d", decpart)
+
+	// Reduce precision according to numPlaces.
+	if len(rightWithTail) > int(numPlaces) {
+		rightWithTail = rightWithTail[0:numPlaces]
+	}
+
+	// Separate trailing zeros.
+	right := strings.TrimRight(rightWithTail, "0")
+	tail := strings.TrimPrefix(rightWithTail, right)
+
+	// For shorter values sent from system
+	amt := strconv.FormatInt(v, 10)
```

Comment:
I mentioned previously that there's no need to format `amt` and check it's length.  You can simply `if intpart == 0`

Reply:
I do not think we are on the same page I am formatting it because I want to get the original input which is a string turning it into an int in the case that the number of inputs passed in is less than 8 digits in which case I assume the intpart decpart wont work.

Reply:
https://play.golang.org/p/T1_9G7qGfro